tests/src/Simple/Threading/Threading.cs

```diff
@@ -758,6 +789,282 @@ public static void MutexMaximumReacquireCountTest()
     }
 }
 
+internal static class ThreadPoolTests
+{
+    [Fact]
+    public static void RunProcessorCountItemsInParallel()
+    {
+        int count = 0;
+        AutoResetEvent e0 = new AutoResetEvent(false);
+        for(int i = 0; i < Environment.ProcessorCount; ++i)
+        {
+            Task.Factory.StartNew(() => {
```

Comment:
Should have most of the tests work against QueueUserWorkItem directly, since these tests intend to test the thread pool and not tasks, except where it is testing something specific to tasks.